src/__tests__/app.test.js

```diff
@@ -0,0 +1,154 @@
+import * as app from '../store/app';
+
+/**
+ * MOCK DATA
+ */
+const appWaitingState = {
+  app: {
+    appWaiting: 1,
+  }
+};
+
+const mobileDrawerOpenedState = {
+  app: {
+    isMobileDrawerOpened: true
+  }
+};
+
+const multipleAlertsState = {
+  app: {
+    alerts: [
+      { desc: 'firstAlert', severity: 'error', title: 'firstAlert' },
+      { desc: 'lastAlert', severity: 'error', title: 'lastAlert' }
+    ],
+  }
+};
+const alert = {
+  desc: 'firstAlert',
+  severity: 'error',
+  title: 'firstAlert'
+};
+const alerts = [
+  { desc: 'firstAlert', severity: 'error', title: 'firstAlert' },
+  { desc: 'lastAlert', severity: 'error', title: 'lastAlert' }
+];
+const lastAlert = {
+  id: 1,
+  desc: 'lastAlert',
+  severity: 'error',
+  title: 'lastAlert'
+};
+
+const removeAlertState = {
+  alerts: [{ desc: 'api success desc', severity: 'error', title: 'API ERROR' }],
+  appWaiting: 0,
+  isMobileDrawerOpened: false
+};
+const appState = {
+  alerts: [],
+  appWaiting: 1,
+  isMobileDrawerOpened: false
+};
+/**
+ * ACTIONS
+ */
+describe('actions', () => {
+  it('toggleMobileDrawer', () => {
+    const expectedAction = {
+      type: app.TOGGLE_MOBILE_DRAWER,
+    };
+    expect(app.toggleMobileDrawer()).toEqual(expectedAction);
+  });
+  it('setAppWaiting', () => {
+    const expectedAction = {
+      type: app.SET_APP_WAITING,
+    };
+    expect(app.setAppWaiting()).toEqual(expectedAction);
+  });
+  it('addAlert', () => {
+    const expectedAction = {
+      type: app.ADD_ALERT,
+      payload: alert
+    };
+    expect(app.addAlert(alert)).toEqual(expectedAction);
+  });
+  it('removeAlert', () => {
+    const expectedAction = {
+      type: app.REMOVE_ALERT,
+      payload: 1
+    };
+    expect(app.removeAlert(1)).toEqual(expectedAction);
+  });
+});
+/**
+ * SELECTORS
+ */
+describe('selectors', () => {
+  it('getIsMobileDrawerOpened', () => {
```

Comment:
I suggest to define a local state over here inside the selector describer, and you do not have to declare a separated app store for each value. And use that as a parameter of a selector.
```javascript
const state = {
  app: {
   ...
  },
};
```

Reply:
I've wanted to differentiate the state for each test, but I also agree with this. Modifying it.